core/modules/cart/store/actions.ts

```diff
@@ -482,78 +380,54 @@ const actions: ActionTree<CartState, RootState> = {
   async applyCoupon ({ getters, dispatch }, couponCode) {
     if (getters.isTotalsSyncEnabled && getters.isCartConnected) {
-      const task = await TaskQueue.execute({ url: config.cart.applycoupon_endpoint.replace('{{coupon}}', couponCode),
-        payload: {
-          method: 'POST',
-          headers: { 'Content-Type': 'application/json' },
-          mode: 'cors'
-        },
-        silent: false
-      })
-      if (task.result === true) {
+      const { result } = await CartService.applyCoupon(couponCode)
+
+      if (result) {
         dispatch('syncTotals', { forceServerSync: true })
       }
-      return task.result
+      return result
     }
-    return null
   },
-  /** authorize the cart after user got logged in using the current cart token */
-  authorize ({ dispatch }) {
-    StorageManager.get('user').getItem('last-cart-bypass-ts', (err, lastCartBypassTs) => {
-      if (err) {
-        Logger.error(err, 'cart')()
-      }
-      if (!config.cart.bypassCartLoaderForAuthorizedUsers || (Date.now() - lastCartBypassTs) >= (1000 * 60 * 24)) { // don't refresh the shopping cart id up to 24h after last order
-        dispatch('connect', { guestCart: false })
-      }
-    })
+  async authorize ({ dispatch }) {
+    const userStorage = StorageManager.get('user')
+    const lastCartBypassTs = await userStorage.getItem('last-cart-bypass-ts')
+
+    if (!config.cart.bypassCartLoaderForAuthorizedUsers || (Date.now() - lastCartBypassTs) >= (1000 * 60 * 24)) {
+      dispatch('connect', { guestCart: false })
+    }
   },
-  /** connect cart to the server and set the cart token */
   async connect ({ getters, dispatch, commit }, { guestCart = false, forceClientState = false }) {
-    if (getters.isCartSyncEnabled) {
-      return _connect({ guestCart, forceClientState }).then(task => {
-        const cartToken = task.result
-        if (task.resultCode === 200) {
-          Logger.info('Server cart token created.', 'cart', cartToken)()
-          commit(types.CART_LOAD_CART_SERVER_TOKEN, cartToken)
-          return dispatch('sync', { forceClientState, dryRun: !config.cart.serverMergeByDefault })
-        } else {
-          let resultString = task.result ? toString(task.result) : null
-          if (resultString && (resultString.indexOf(i18n.t('not authorized')) < 0 && resultString.indexOf('not authorized')) < 0) { // not respond to unathorized errors here
-            if (_connectBypassCount < config.queues.maxCartBypassAttempts) {
-              Logger.log('Bypassing with guest cart' + _connectBypassCount, 'cart')()
-              _connectBypassCount = _connectBypassCount + 1
-              Logger.error(task.result, 'cart')()
-              return dispatch('connect', { guestCart: true })
-            }
-          }
-        }
-      })
-    } else {
-      Logger.warn('Cart sync is disabled by the config', 'cart')()
-      return _getDifflogPrototype()
+    if (!getters.isCartSyncEnabled) return
```

Comment:
My idea was that all the actions that modifies cart returns difflog :)

Reply:
@pkarw yeah, one question about that - do we need that `difflog` everywhere? because even some fields from `difflog` are not used anywhere (such as server responses). Should I keep that?

Reply:
yeah, the `diffLog` is covering pretty much use cases at once; we probably don't need to use it everywhere but any action that calls `sync` should return a `diffLog` i believe - otherwise, there is no proper way to handle possible issues/status changes in the caller